### ISSUE TITLE
Take Action Page / Tasks Block

### ISSUE THREAD

**Issue by viniromualdo** [greenpeace/planet4-presentation-layer#106]
Colors have been updated according to official palette. 
Latest file is up in Drive ([take_action_all_screens_light_v2](https://drive.google.com/open?id=0B6J-QWxbAW55ZV9IdUJyeDNrTWM))

**Interaction behavior**
S + M screens: works like an accordion, so default starts with the first task expanded, all rest collapsed. Tapping on collapsed task causes it to expand and reveal inner paragraph and Call to action (btn, img...). The previously expanded tasks remain expanded unless intentionally collapsed back. Style should change accordingly as in the designs.

L + XL screens: all content is always visible in all tasks. By default the first task has a 100% opacity (even when not hovered). Hovering over another task causes it to go from faded state (50% opacity) to 100% opacity, while all other tasks get faded. Mouse cursor should be normal everywhere in the task area EXCEPT when over a CTA (btn, img...). In this last case cursor should be a hand. There are no longer click states, in other words, clicking on a task doesn't change its color.

I'm tagging @ashish173 who's been working on it, but also @mcshanea since I couldn't find Shannon's username and I'm not sure who's working on the interaction itself.

Any questions please ping me.

**Comment by Kluny** [comment]
@Kluny here. Is @ashish173 still trying to make it work?

**Comment by viniromualdo** [comment]
@Kluny , I believe he was finishing styling it, but having difficulties to make it work with a single markup.
I need to know if this is achievable or not, so I can redesign S + M interactions if necessary.
Is someone helping @ashish173 , @mcshanea ?

**Comment by ashish173** [comment]
@viniromualdo Yes, I have been working on the larger screens styling, had a chat with @Kluny yesterday and agreed that she'd continue from where I leave today.

**Comment by viniromualdo** [comment]
Fantastic, thanks for letting me know and please keep me posted

**Comment by Kluny** [comment]
I made a little progress with the medium template. The mobile/small one looks really good. 

Question: in the AI file there's a yellow "share" button under the steps, in the current html this isn't there and there's an orange "add this to my action list" button instead. Could you confirm that the yellow "Share" button is what we want? 

I had some more questions about step visibility behaviour, but I just realized you answered them already. There's a fair bit more work to do but nothing impossible! I'll be afk tomorrow but I can put in some hours over the weekend.

**Comment by mikeselander** [comment]
@Kluny can you please update this ticket with some progress information and estimations?

**Comment by Kluny** [comment]
From my notes on the PR:

Broke @viniromualdo's spec down into this task list:

√ S + M screens: works like an accordion, so default starts with the first task expanded,

√ all rest collapsed.
√ Tapping on collapsed task causes it to expand and reveal inner paragraph and Call to action (btn, img...).
√ The previously expanded tasks remain expanded unless intentionally collapsed back.

TODO: Style should change accordingly as in the designs.
    (change "take action" button to "share" and orange to yellow)

TODO: First task number should be bigger than the others

√ L + XL screens: all content is always visible in all tasks.

TODO: Hovering over another task causes it to go from faded state (50% opacity) to 100% opacity, while all other tasks get faded. (Right now first task stays opaque)

√ By default the first task has a 100% opacity (even when not hovered).
√ Mouse cursor should be normal everywhere in the task area EXCEPT when over a CTA (btn, img...).
√ In this last case cursor should be a hand.
√ There are no longer click states, in other words, clicking on a task doesn't change its color.

I'm still not sure the hover/opacity behaviour is working as it should. Technically I think it meets the spec, but maybe @viniromualdo could look at how it is now and tell me whether it falls short. Should the "active" state be persistent after user mouses off a step?

**Comment by viniromualdo** [comment]
@Kluny functionally on mobile works great, but it seems to me that you've been looking at an outdated file in terms of styling. You should be looking at the latest one I've initially pointed above (take_action_all_screens_light_v2), so there's a few style updates there. 
![tasks](https://user-images.githubusercontent.com/12191933/34197205-b65676c0-e565-11e7-8cb7-2bea86559884.jpg)

From the checklist you added above: 
#3 still needs to be done, 
#4 it should just be about styling this block according to latest designs,
#5 this behavior only happens in M(tablet) screensize
![tasks2](https://user-images.githubusercontent.com/12191933/34197991-fd557cda-e568-11e7-82a7-3c52fdbcdde7.jpg)
Basically, it still works like an accordion on M, so the active task is 100% opacity and has a bigger number + expanded text content bellow, while its call to action content floats on its right (i.e. imgs, btns). The collapsed tasks are 50% opacity (entirely, not just their number), their numbers "bubble" are smaller in size and when any area of a collapsed task is pressed it becomes active, while previously active task collapses. A smooth transition of the number "bubble" inflating/deflating would be very nice to have.

On L and XL screens, 
Points 7, 9 and 11 on your checklist aren't done properly.
There's seems to be a confusion when I say the task goes to 50% opacity I'm referring to it entirely (whole content area), not just its number.
![tasksl](https://user-images.githubusercontent.com/12191933/34199060-bd4a7448-e56c-11e7-958c-a8006307f412.jpg)

The active state only exists in one task at a time. So, for example, if user hovers over task 3 last and scrolls down the page, active will be persistent on task 3 until user hovered over another task.

Also it's important to note that tasks are up to 4, not necessarily always 4. So if there's only 2, for example, they should span across the container accordingly.

I hope I'm giving you feedback on the actual updated page your worked on. ([this](https://greenpeace.github.io/planet4-presentation-layer/) is what I looked at)

Any questions we can skype later tonight, it's so much easier to talk these things through rather than writing it.

cc @tolen

**Comment by Kluny** [comment]
Thanks @viniromualdo! The page you linked to (https://greenpeace.github.io/planet4-presentation-layer/) is not up to date with what's in the current pull request, so we have progressed beyond that, but that feedback and the screenshots are helpful anyway. 

Are you able to pull the current branch and run it locally to see the same thing as what I'm looking at? If not I'll post screencaps in a bit.

**Comment by Kluny** [comment]
Here's where we're at now. It's not what I'd like it to be, but it's closer. I focussed on functionality while trying to incorporate as many as possible of those late design changes. 

(30 second video) https://imgur.com/a/pGUMo
